### PR TITLE
Added mulitgraph redis plugin

### PR DESCRIPTION
**Pull request by rwky**
This is a fork of the redis_ plugin

Changes in the fork:

* Supports multiple redis instances
* Supports AWS elasticache and other environments where the CONFIG
command is disabled
* Uses multigraph to separate graphs by instance

### CODE REVIEW

**Comment by brknkfr** [comment]
I have multiple redis instances on my machine, each on a different socket. When I specify multiple network connections, all with 127.0.0.1, but a different port, the plugin won't work.

And somehow when specifying multiple socket paths it won't work neither.

File: plugins/redis/redis [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+## 3. Edit plugin-conf.d/munin-node with the options to connect to your redis instances (see below for an example)
+## 4. Restart munin-node service
+##
+## Example config
+## [redis]
+## env.host1 127.0.0.1
+## env.port1 6379
+## env.password1 password
+## env.title_prefix1 redis-1
+## env.host2 /run/redis.sock
```
[the remark this answers is not in the document]
Comment (rwky) [reply plugins/redis/redis]: Yep fixed that! Good spot.

**Comment by rwky** [comment]
> I have multiple redis instances on my machine, each on a different socket. When I specify multiple network connections, all with 127.0.0.1, but a different port, the plugin won't work. Multiple hosts will work, but not multiple unix sockets.

Can you please paste your env settings so I can debug?

> Maybe change the plugin in a way that you can specify an instance with ln -s redis_ redis_instancename than specifying multiple env entries in config?

That's an option but I don't see a massive benefit since the env entries will still need to be added, the existing solution seems simpler IMHO.

**Comment by brknkfr** [comment]
> > I have multiple redis instances on my machine, each on a different socket. When I specify multiple network connections, all with 127.0.0.1, but a different port, the plugin won't work. Multiple hosts will work, but not multiple unix sockets.
> 
> Can you please paste your env settings so I can debug?
> 
This doesn't work:
```
[redis]
env.unixsocket1 /var/run/redis/redis-cache_6379.sock
env.password1 VERYSECURE
env.title_prefix1 cache_6379
env.unixsocket2 /var/run/redis/redis-sessions_6380.sock
env.password2 VERYSECURE
env.title_prefix2 cache_6380
```
This doesn't work neither:
[redis]
env.host1 127.0.0.1
env.port1 6379
env.password1 VERYSECURE
env.title_prefix1 cache_6379
env.host2 127.0.0.1
env.port2 6380
env.password2 VERYSECURE
env.title_prefix2 cache_6380

> > Maybe change the plugin in a way that you can specify an instance with ln -s redis_ redis_instancename than specifying multiple env entries in config?
> 
> That's an option but I don't see a massive benefit since the env entries will still need to be added, the existing solution seems simpler IMHO.

**Comment by rwky** [comment]
This is my config which works

```
[redis]
env.host1 redis.domain.com
env.port1 6379
env.title_prefix1 dev-cache
env.host2 redis2.domain.com
env.port2 6379
env.title_prefix2 dev-sessions
```

It's almost the same as your second one. So I'm not sure why it doesn't work, can you please enable debugging and check your logs for any hints on what's going wrong?

The unixsocket config is clearly broken I'll sort that

**Comment by brknkfr** [comment]
> It's almost the same as your second one. So I'm not sure why it doesn't work, can you please enable debugging and check your logs for any hints on what's going wrong?
> 
I re-entered my config again and now it's working with multiple instance on the same host but different port. I probably messed up something with the order of the parameters.

**Comment by rwky** [comment]
Grand, glad it's sorted. I'll sort out the unixsocket thing after work :)

**Comment by rwky** [comment]
I've fixed the unix socket issues, the host environment option now accepts unix sockets

**Comment by sumpfralle** [comment]
Sorry for letting you wait for too long :(

I appreciate your improvements for the redis plugin.

I have some minor questions following up on your discussion above:
* Could you please reformat the documentation to the perlpod format?
* Do you think, that the new plugin offers all the features of the old plugin? In this case I would recommend, that you remove the old plugin and refer to the new one (in the commit message).

**Comment by rwky** [comment]
@sumpfralle done all those! Let me know if you need anything else tweaked.

**Comment by sumpfralle** [comment]
it look great - thanks!

**Pull request merged**